pmlc/dialect/pxa/transforms/tile.cc

```diff
@@ -31,11 +32,23 @@ void performTiling(AffineParallelOp op, llvm::ArrayRef<int64_t> tileSizes) {
   // Make the inner parallel for (abve all other code);
   auto inner = builder.create<AffineParallelOp>(
       op.getLoc(), op.getResultTypes(), lbMap, outerIdxs, ubMap, outerIdxs);
-  // Splice instructions into the interior
+  // Build a map from old indices to new indices
+  mlir::BlockAndValueMapping mapper;
+  auto innerIdxs = inner.getBody()->getArguments();
+  for (unsigned i = 0; i < outerIdxs.size(); ++i) {
+    mapper.map(outerIdxs[i], innerIdxs[i]);
+  }
+  // Clone new instructions into the interior
   auto &innerLoopOps = inner.getBody()->getOperations();
   auto &outerLoopOps = outerBody->getOperations();
-  innerLoopOps.splice(std::prev(innerLoopOps.end()), outerLoopOps,
-                      std::next(outerLoopOps.begin(), 1), outerLoopOps.end());
+  for (auto& origOp : outerLoopOps) {
+    if (&origOp == inner.getOperation()) {
+      continue;
+    }
+    Operation* newOp = origOp.clone(mapper);
+    innerLoopOps.push_back(newOp);
+  }
+  outerLoopOps.erase(std::next(outerLoopOps.begin(), 1), outerLoopOps.end());
```

Comment:
I think we can use `Region::cloneInto` here.

Reply:
I tried Region::clone. It appends the src blocks after the dest region blocks. In our case, the entry block, bb0, in inner loop still exists. We have to remove bb0 and reset bb1's arguments and adjust cfg. It's not that convenient.